tests/pyccel/test_pyccel.py

```diff
@@ -508,3 +509,63 @@ def test_print_strings(language):
 def test_print_sp_and_end(language):
     types = str
     pyccel_test("scripts/print_sp_and_end.py", language=language, output_dtype=types)
+
+def test_headers(language):
+    test_file = "scripts/test_headers.py"
+    test_file = os.path.normpath(test_file)
+    test_file = get_abs_path(test_file)
+
+    header_file = 'scripts/headers.pyh'
+    header_file = os.path.normpath(header_file)
+    header_file = get_abs_path(header_file)
+
+    with open(test_file, 'w') as f:
+        code = ("from headers import f\n"
+                "def f(x):\n"
+                "    y = x\n"
+                "    return y\n"
+                "print(f(1))\n")
+
+        f.write(code)
+
+    with open(header_file, 'w') as f:
+        code =("#$ header metavar ignore_at_import=True\n"
+               "#$ header function f(int)")
+
+        f.write(code)
+
+    test_file = os.path.normpath(test_file)
+    cwd = os.path.dirname(test_file)
+    cwd = get_abs_path(cwd)
+
+    pyccel_commands = " --language="+language
+
+    compile_pyccel(cwd, test_file, pyccel_commands)
+
+
+    lang_out = get_lang_output(get_exe(test_file))
+    assert int(lang_out)
```

Comment:
Shouldn't we be checking the type of `lang_out` to make sure that the right type is used?

Reply:
Never mind. This is a string anyway. The test comes from the compilation